### ISSUE TITLE
Handling situation when user inserts option in other's option value (eg. app command --option1 --option2)

### ISSUE THREAD

**Issue by matiduck** [inausoft/netCLI#14]
I wrote a simple Console app:
```c#
using System;

namespace ConsoleApp4
{
    using inausoft.netCLI;
    using Microsoft.Extensions.DependencyInjection;
    using System;

    namespace ConsoleApp1
    {
        class Program
        {
            static int Main(string[] args)
            {
                var services = new ServiceCollection();
                services.AddCLI(config => {
                    config.Map<MoveCommand, MoveCommandHandler>();
                });

                var provider = services.BuildServiceProvider();

                return provider.RunCLI(args);
            }
        }

        [Command("move", "Moves files from one path to another.")]
        class MoveCommand
        {
            [Option("pathFrom", "Current files locations.")]
            public string From { get; set; }

            [Option("pathTo", "Destination path for the files.")]
            public string To { get; set; }

            [Option("force", "Overrides if files exists.")]
            public bool IsForce { get; set; }
        }

        class MoveCommandHandler : CommandHandler<MoveCommand>
        {
            public override int Run(MoveCommand options)
            {
                Console.WriteLine($"Files moved from {options.From} to {options.To}");

                return 0;
            }
        }
    }
}
```

then I run this program with following arguments: 
```
.\ConsoleApp4.exe move --pathFrom sourcePath --pathTo --SomeInvalidOption
```
O clearly make a wrong usage since I accidently forgot to type `pathTo` options value, but as a result i got following exception stack trace: 

```
.\ConsoleApp4.exe : Unhandled exception. System.ArgumentException: Object of type 'System.Boolean' cannot be converted to type 'System.String'.
At line:1 char:1
+ .\ConsoleApp4.exe move --pathFrom sourcePath --pathTo --SomeInvalidOp ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (Unhandled excep...System.String'.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
   at System.RuntimeType.TryChangeType(Object value, Binder binder, CultureInfo culture, Boolean needsSpecialCast)
   at System.RuntimeType.CheckValue(Object value, Binder binder, CultureInfo culture, BindingFlags invokeAttr)
   at System.Reflection.MethodBase.CheckArguments(Object[] parameters, Binder binder, BindingFlags invokeAttr, CultureInfo culture, Signature sig)
   at System.Reflection.RuntimeMethodInfo.InvokeArgumentsCheck(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at inausoft.netCLI.Deserialization.RegexOptionsDeserializer.Deserialize(Type type, String optionsExpression) in C:\Users\kaczo\Source\Repos\officialNetCLI\src\inausoft.netCLI\Deserialization\RegexOptionsDeserializer.cs:line 59
   at inausoft.netCLI.Deserialization.RegexOptionsDeserializer.Deserialize(Type type, String[] args) in C:\Users\kaczo\Source\Repos\officialNetCLI\src\inausoft.netCLI\Deserialization\RegexOptionsDeserializer.cs:line 20
   at inausoft.netCLI.Executor.RunCLI(IServiceProvider serviceProvider, String[] args) in C:\Users\kaczo\Source\Repos\officialNetCLI\sr
```

I believe that we should provide more meaningful information when this situation happens

**Comment by jmarlew** [comment]
Interesting, I would expect that `--SomeInvalidOption` would be treated as value for `pathTo` and eventually fail later eg. because this path does not exist.
By looking at this I don't understand this error at all.
Is that behaviour expected?
What if my option expects a value that looks like `--something` - this is a strange use case, agreed, but should it be allowed?

**Comment by MarcinIN** [comment]
This is expected since no value options are treated as `bool`.
For example: `move --PathTo sample\path --force` for code below
```cs
[Command("move")]
public MoveCommand
{
    [Option("force")]
    public bool IsForced { get; set; }
}

...
public int Run(MoveCommand command)
{
     if(command.IsForced)
     {
           //do stuff
     }
}
...
```
would simplify usage from `--force true` to simple `--force`

**Comment by MarcinIN** [comment]
But going back to main problem I guess we would need to wrap this exception with InvalidOptionException.

**Comment by MarcinIN** [comment]
Fixed with #23 PR.
